### PR TITLE
MM-29663 - Summary and Tasks tabs for RHS Details View

### PR DESCRIPTION
**Pull request by cpoile**
#### Summary
- This is the first part of the epic, setting up the summary and tasks tabs and splitting up the existing components into their proper tabs.
- After discussion with @itao:
  - the `current stage` component should be in both tabs, but the `commander` and `duration` components should only be in the summary tab.
  - we should remember what tab that particular incident is on. But it's okay to forget that stage on a refresh (just like the webapp forget's all RHS state on a refresh).
    - that's why the channelId -> RHSTabState is kept in redux.
- Looks like:
image
image

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-29663

### CODE REVIEW

File: webapp/src/components/rhs/rhs_incident_tasks.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+                                            dispatch(toggleRHS());
+                                        }
+                                    }}
+                                />
+                            ))}
+                        </div>
+                    </InnerContainer>
+                </div>
+            </Scrollbars>
+            <RHSFooterTasks incident={props.incident}/>
```
Comment (cpoile) [inline webapp/src/components/rhs/rhs_incident_tasks.tsx:88]: refactored to clean this file up a bit

File: webapp/src/components/rhs/rhs_shared.tsx [hunk 2]
```diff
@@ -5,11 +5,10 @@ import React from 'react';
 import styled from 'styled-components';
 
 export const RHSContainer = styled.div`
-    height: calc(100vh - 120px);
+    height: calc(100vh - 119px);
```
Comment (cpoile) [inline webapp/src/components/rhs/rhs_shared.tsx:8]: for some reason this was leaving a pixel open at the bottom of the footer, you could see the scrollbar and the very bottom

File: webapp/src/components/rhs/rhs_shared.tsx [hunk 3]
```diff
@@ -5,11 +5,10 @@ import React from 'react';
...
     display: flex;
     flex-direction: column;
     position: relative;
-    overflow: auto;
```
Comment (cpoile) [inline webapp/src/components/rhs/rhs_shared.tsx:12]: this was leaving a permanent scrollbar on the whole rhs

**Comment by cpoile** [comment]
Yes, I know e2e tests are failing -- fixing them up. :)

File: webapp/src/components/rhs/rhs_footer_tasks.tsx [hunk 4]
```diff
@@ -0,0 +1,163 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
```
Comment (cpoile) [inline webapp/src/components/rhs/rhs_footer_tasks.tsx:1]: refactored, only minor changes

File: webapp/src/components/rhs/stage.tsx [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+    &:last-child {
+        border-top-right-radius: 4px;
+        border-bottom-right-radius: 4px;
+        margin-right: 0;
+    }
+`;
+
+interface StageProps {
+    incident: Incident;
```
Comment (cpoile) [inline webapp/src/components/rhs/stage.tsx:47]: this is the main change

File: webapp/src/components/rhs/stage.tsx [hunk 6]
```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
```
Comment (cpoile) [inline webapp/src/components/rhs/stage.tsx:1]: refactored, changes to props to simplify

**Review by crspeller — APPROVED** [review]
Looks good!

**Review by itao — APPROVED** [review]
Nice LGTM!

**Review by agarciamontoro — APPROVED** [review]
Man, I have *nothing* to say, it's almost frustrating :joy: Great work!

**Comment by cpoile** [comment]
Sorry, force pushed because had to rebase on all the changes to master.

**Pull request merged**